### PR TITLE
more logging around main image

### PR DESCRIPTION
**Pull request by bjfletcher**
to understand better the disappearing image problem on frontpage

### CODE REVIEW

File: server/models/article-v3.js [hunk 1]
```diff
@@ -32,14 +32,55 @@ function fetchCapiV2 (articleId) {
...
+						mainImageId: mainImageId,
+						imageSetMemberId: imageSetMemberId,
+						isImageDefined: !!image,
+						imageBinaryUrl: image ? image.binaryUrl : null
+					});
+					return image;
+
+				})
+				.catch(err => {
+					logger.warn(Object.assign({}, err, {
```
Comment (bjfletcher) [inline server/models/article-v3.js]: @matthew-andrews what do you think about this way of flattening err object along with some more details, for the logger?

**Comment by matthew-andrews** [comment]
looks good! +1

File: server/models/article-v3.js [hunk 2]
```diff
@@ -32,14 +32,55 @@ function fetchCapiV2 (articleId) {
...
+						event: 'MAIN_IMAGE_FETCH_FAIL',
+						mainImageId: mainImageId,
+						imageSetMemberId: imageSetMemberId
+					}));
```
Comment (ironsidevsquincy) [inline server/models/article-v3.js]: You _should_ be able to just do

```
logger.warn(err, { event: ... });
```

File: server/models/article-v3.js [hunk 3]
```diff
@@ -32,14 +32,55 @@ function fetchCapiV2 (articleId) {
...
 		})
 
 		// Return undefined so that the caller (which checks whether this output is false)
 		// doesn't wrongly add an empty `mainImage` object to the article model.
-		.catch(() => {});
+		.catch(err => {
+			logger.warn(Object.assign({}, err, {
+				event: 'MAIN_IMAGE_SET_FETCH_FAIL',
+				mainImageId: mainImageId
+			}));
```
Comment (ironsidevsquincy) [inline server/models/article-v3.js]: ditto

**Comment by bjfletcher** [comment]
My very many thanks, @matthew-andrews & @ironsidevsquincy ... pushing this and then will work with editorial team to identify disappearing images on frontpage with hopefully better logging

**Pull request merged**